### PR TITLE
iterators task (1) - Pysarenko

### PR DESCRIPTION
**Pull request by Elzianor**

### CODE REVIEW

File: iterators_task_1/student/src/impl.cpp [hunk 1]
```diff
@@ -1,2 +1,18 @@
 #include "impl.h"
 
+std::string reverseWords(const std::string& input)
+{
+    std::stringstream stream{input};
+
+    std::vector<std::string> parsed{std::istream_iterator<std::string>(stream),
+                std::istream_iterator<std::string>()};
+
+    stream = std::stringstream();
```
Comment (smay1613) [inline iterators_task_1/student/src/impl.cpp:10]: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es26-dont-use-a-variable-for-two-unrelated-purposes

File: iterators_task_1/student/src/impl.cpp [hunk 2]
```diff
@@ -1,2 +1,18 @@
...
+    std::reverse_copy(parsed.begin(), parsed.end(), std::ostream_iterator<std::string>{stream, " "});
+
+    std::string merged{stream.str()};
+    if(merged.length())
```
Comment (smay1613) [inline iterators_task_1/student/src/impl.cpp:14]: empty()

File: iterators_task_1/student/src/impl.cpp (hunk 1 above)
Comment (Elzianor) [reply iterators_task_1/student/src/impl.cpp]: Fixed

File: iterators_task_1/student/src/impl.cpp (hunk 2 above)
Comment (Elzianor) [reply iterators_task_1/student/src/impl.cpp]: Fixed

**Pull request closed without merging**